app/models/immunization.rb

```diff
@@ -0,0 +1,79 @@
+class Immunization
+  class << self
+    @tradenames = nil
+
+    def tradenames
+      @tradenames ||= load_tradenames
+    end
+
+    def load_tradenames
+      f = File.open(Rails.root.join('app', 'assets', 'iisstandards_tradename.xml'))
+      @tradenames = Nokogiri::XML(f)
+    end
+  end
+
+  def initialize(fhir_immunization)
+    @fhir_immunization = fhir_immunization
+    @occurrence = occurrence
+    @vaccine = vaccine
+    @lot_number = lot_number
+    @vaccinator = vaccinator
+  end
+
+  def attributes
+    {
+      'occurrence' => @occurrence,
+      'vaccine' => @vaccine,
+      'lot_number' => @lot_number,
+      'vaccinator' => @vaccinator
+    }
+  end
+
+  def occurrence
+    from_fhir_time(@fhir_immunization.occurrenceDateTime)
```

Comment:
This method was copy and pasted in two places for each model. I recommend created a module for shared behavior for these view models

Reply:
moved to helper module